### DISCUSSION TITLE
8 bits per pixel grayscale bmp becomes 24 bits per pixel bmp after extending it.

### DISCUSSION THREAD

**Discussion by EvilOctopusFu** [ImageMagick/ImageMagick#4439] [Help] [answered]
I use this command to extend the image.
`convert in.bmp -background black -gravity center -extent 258x258 BMP3:out.bmp`
The output image will become 24 bits per pixel, and given that my extended border color is black, I want to keep it as an 8-bit image with a 256-color palette.

```
identify in.bmp
in.bmp BMP3 256x256 256x256+0+0 8-bit sRGB 256c 66614B 0.000u 0:00.000

identify out.bmp
out.bmp BMP3 258x258 258x258+0+0 8-bit sRGB 200262B 0.000u 0:00.001
```

I have tried using `-colors 256`to limit the number of colors, but this causes the number of colors in the output image to be significantly less than 256 colors, which causes degradation of image quality.
This can be clearly seen in a gradient color image.
```
magick -size 256x256 gradient: linear_gradient.bmp
convert linear_gradient.bmp -background black -gravity center -extent 258x258 -colors 256 BMP3:out.bmp
```

The version of the ImageMagick I am using is “ImageMagick 7.1.0-13 Q16-HDRI x64 2021-10-29”

[image.zip](https://github.com/ImageMagick/ImageMagick/files/7501457/test_image.zip)

**Comment by snibgo** [comment]
Please show in.bmp so we can try to reproduce the problem.

**Reply by EvilOctopusFu** [reply]
You can use a gradient image generated by magick as the input image.
`magick -size 256x256 gradient: in.bmp`

**Comment by snibgo** [comment]
```
magick -size 256x256 gradient: in.bmp
```
For me with IM v7.1.0-4, that gives a non-palette file, with 24 bits per pixel. Using "-type palette" makes a palette output.

**Reply by EvilOctopusFu** [reply]
![in](https://user-images.githubusercontent.com/39089154/140451089-7107de69-b311-4cd0-beb2-ce9fca958981.jpg)
Because of github's limitations, I changed the file extension to jpg, but it's actually a bmp file.

**Reply by fmw42** [reply]
In the future, just zip your BMP and you can upload zip format to GIThub

**Comment by EvilOctopusFu** [comment, marked as answer]
End up solving the problem by turn off the dithering, using the `-dither None` option.
`magick in.bmp -background black -gravity center -extent 258x258 -dither None -compress None -colors 256 BMP3:out.bmp`
